packages/hdwallet-ledger/src/ledger.ts

```diff
@@ -318,12 +318,22 @@ export class LedgerHDWallet implements core.HDWallet, core.BTCWallet, core.ETHWa
   }
 
   public async getDeviceID (): Promise<string> {
-    const { device: { deviceID }} = this.transport as any
+    const { device: { serialNumber: deviceID }} = this.transport as any
     return deviceID
   }
 
   public async getFeatures (): Promise<any> {
-    return this.transport.getDeviceInfo()
+    await this.transport.open()
+    try {
+      return await this.transport.getDeviceInfo()
```

Comment:
wouldn't the early return prevent us from closing the transport in the final block ?

Reply:
I thought `finally`s ran after `return`s?

Reply:
```
function tryit() {
  try {
    return console.log("try")
  } finally {
    console.log("finally")
  }
}
tryit()
```
prints:
```
try
finally
```

Reply:
looks like you are right. I was trying it here too:
https://repl.it/@dviramontes/DistantRecklessKernelmode

Reply:
disregard @kaladinlight :)